providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/StorageStateDTO.java

```diff
@@ -0,0 +1,22 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.storage;
+
+
+import dev.openfeature.contrib.providers.flagd.resolver.process.model.FeatureFlag;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.Map;
+
+@Getter
+@ToString
+@EqualsAndHashCode
+public class StorageStateDTO {
+    private final StorageState storageState;
+    private final Map<String, FeatureFlag> changedFlags;
```

Comment:
I am not sure how we handle this in other SDKs, but do we need the whole "FeatureFlag" object here, or is a list of Keys enough - it seems to me we do a lot of mapping - around this map, for the fact that only the key is relevant to us. We can reduce the complexity of the rest of the code if we only store the key of the flags in a list here (or if I am missing something). The only thing that might be interesting for the changed flags is the actual manipulation, if added, removed, or updated. wdyt?

Reply:
That's right, we have a method in the storage interface that retrieves the feature flag using its key, allowing us to access it from anywhere whenever needed.